File: Incubation/tags/EDSN_JUNE_2013/psim/src/org/space/oss/psim/ConfigByMap.java
package org.space.oss.psim;

import java.util.Map;
import java.util.TreeMap;

public class ConfigByMap 
    implements Config 
{
	protected Map<String,String> values_;
	protected String[] args_;
	
	public ConfigByMap(String args[])
	{
	    values_ = new TreeMap<String,String>();
	    args_ = args;
	}
	
	public void addValue(String name,String value)
	{
	    values_.put(name, value);	
	}
	
	public String getValue(String name) 
	{
		return values_.get(name);
	}	
}

File: Incubation/trunk/psim-scala/src/main/scala/org/space/oss/psim/ConfigByMap.scala
package org.space.oss.psim

import scala.collection.mutable.HashMap

class ConfigByMap extends Config {
	val values = new HashMap[String,String]()
	val beanTypes = new HashMap[String,String]()
	
	override def getValue(name: String): Option[String] = values.get(name)
	
	override def setBeanType(name:String, btype:String) { beanTypes += (name -> btype)}
	
	override def getBean(name:String): Any = {
	  beanTypes.get(name) match {
	    case Some(typeName) => Class.forName(typeName).newInstance
	    case None => null
	  }
	}
}